journal/src/main/java/io/camunda/zeebe/journal/file/Segment.java

```diff
@@ -230,6 +235,21 @@ private void safeDelete() {
     }
   }
 
+  boolean flush() {
+    try (final var ignored = metrics.observeSegmentFlush()) {
+      buffer.force();
+    } catch (final UncheckedIOException e) {
+      if (isOpen()) {
+        throw e;
+      }
+
+      LOG.trace("Flushing failed on a closed or deleted segment, and will be ignored");
+      return false;
```

Comment:
I debated if we should return true or not. It will be mostly if we update the meta store. I'm still not sure honestly :thinking: If we do, then we can omit the return value and assume a non-exceptional call always succeeds.

Reply:
I think we can keep it. It might be unnecessary, but I feel safe to keep it incase if there is an edge case that we haven't thought about.